src/calculateAverage.js

```diff
@@ -37,7 +37,71 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    let sumOfValues = 0;
+    let countNotExistValues = 0;
+    let isAtLeastOneProp = false;
+    const processingAccumulator = (callback) => {
+      sumOfValues = this.reduce(
+        (sum, user, index, array) => {
+          return sum + callback(user, index, array);
+        }
+        , 0);
+
+      return sumOfValues / this.length;
+    };
+
+    if (typeof options === 'string'
+        || options === null
+        || Array.isArray(options)) {
+      return;
+    }
+
+    if (options === undefined || options.length === 0) {
+      sumOfValues = this.reduce(
+        (sum, item) => sum + item, 0
+      );
+
+      return sumOfValues / this.length;
+    }
+
+    if (typeof options === 'object') {
+      if (Object.keys(options) === 0) {
+        return;
```

Comment:
will be great to move all negative checks to one place in code, then focus on processing positive cases